pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+and a virtual machine begins its execution.
+
+Notice that you cannot use both `commands` and `cmd_file` when defining a job.
+
+#### Example of a cmd_file block
+
+    jobs:
+      - name: A name for this job
+         cmd_file: GitHub/file/plain_text_file
+
+
+### Example of Jobs
+
+The general structure of a job when using the `commands` property is as follows:
+
+    jobs:
+       - name: A name for this job
+          commands:
+             - Command 1
+             - Command 2
+
+The general structure of a job when using the `cmd_file` property is as follows:
+
+    jobs:
+      - name: A name for this job
+         cmd_file: path/to/file/plain_text_file
+
+
+## Prologue and Epilogue
+
+
+A `.semaphore/semaphore.yml` configuration file can have a single `prologue`
+and a single `epilogue` property per `task` property. Both `prologue` and
+`epilogue` properties are optional.
+
+
+### The Prologue property
+
+A `prologue` block is used when you want to execute some commands prior to
+a job. This is usually the case with initialization commands that install
+software, start or stop a service, etc.
+
+Please notice that a project *will fail* if a command in the `prologue` fails to
+execute for some reason.
+
+### Example of prologue
+
+      prologue:
+          commands:
+            - checkout
+
+### The Epilogue property
+
+An `epilogue` block should be used when you want to execute some commands
+after a job has finished, either successfully or unsuccessfully.
+
+Please notice that a project *will not fail* if one or more commands in the
+`epilogue` fail to execute for some reason.
+
+### Example of epilogue
+
+      epilogue:
+          commands:
+            - echo $SEMAPHORE_JOB_RESULT
+            - echo $SEMAPHORE_PIPELINE_ID
+
+## The secrets property
+
+The `secrets` property is for using existing `secrets` environment variables.
+
+Please notice that when the names of the environment variables of two
+more more `secrets` are the same, then the environment variable will
+have the value that can be found in the `secrets` property that was
+imported last.
+
+Additionally, if you try to use a `name` value that does not exist, the
+pipeline will fail to compile and therefore run.
+
+
+### name
+
+The `name` property is compulsory is a `secrets` block because it
+specifies the `secrets` bucket that you want to use. Please note
+that the specified `secrets` bucket should be under the
+active organization.
+
+
+### env_var_names
+
+The `env_var_names` property lets you select the environment variables
+that you want to use from the specified `secrets` bucket. Currently only
+the `[]` value is accepted, which loads all environment variables found under
+the specified  `secrets` bucket.
+
+
+### Example of secrets
+
+      secrets:
+        - name: mysql-secrets
+          env_var_names: []
+
+Note that the environment variable defined in a `secrets` property are used
+like regular environment variables defined in a `env_vars` block.
```

Comment:
Consider dropping `Note` to communicate something. It sounds a bit threatening like something bad is going to happen.

> Environment variables defined in a `secrets` are used like regular environment variables defined in a `env_vars` block.

Reply:
Done!